test/integration/dialects/mysql/connector-manager.test.js

```diff
@@ -87,8 +87,11 @@ if (dialect === 'mysql') {
         .then(connection => {
           // Save current connection
           conn = connection;
-          // simulate a unexpected end
-          connection.close();
+          // simulate a unexpected end from MySQL2
+          connection._protocolError = new Error('Connection lost: The server closed the connection.');
+          connection._protocolError.fatal = true;
+          connection._protocolError.code = 'PROTOCOL_CONNECTION_LOST';
+          connection._notifyError(connection._protocolError);
```

Comment:
@felixfbecker The `validate` method of the mysql connection manager uses these protected properties.  The validate method is also called before acquisition of the connection.  Maybe set a fixed version in the `package.json`?

[lib/dialects/mysql/connection-manager.js#L184](https://github.com/sequelize/sequelize/blob/master/lib/dialects/mysql/connection-manager.js#L184)
```javascript
validate(connection) {
    return connection && connection._fatalError === null && connection._protocolError === null && !connection._closing;
  }
```